clang/lib/Driver/Compilation.cpp

```diff
@@ -99,7 +99,7 @@ static void HandleXarchArgs(DerivedArgList *OffloadArgList, const Driver &D,
               OffloadArgList->MakeArgStringRef(XarchVecs[Index]));
       }
     } else
-      XarchValueRefs.push_back(StringRef(XarchV));
+      XarchValueRefs.push_back(OffloadArgList->MakeArgString(XarchV));
```

Comment:
Any way we can capture the bad behavior in a test other than running through the sanitizer?

Reply:
Hi, @mdtoguchi 
The sycl-xarch.cpp test has covered this line of code already but it just passed due to good luck. This behavior is a general problem, a temporary value is still used when it is out of scope. Sometimes, the memory of this value is not touched even when it is out of scope, this is whey this test happened to pass. In some platform, the memory of this temporary value may be modified, then we will see the lit test fails.
It is very difficult to catch such kind of errors because we can't make the temporary value's memory to be changed when it is out of scope and memory sanitizer is invented to detect such bugs via shadow memory mechanism.
Thanks very much.